TODO: use
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,18)
 #include <linux/config.h>
 #else
instead of just removing config.h
diff -Naur madwifi-0.9.2/ath/if_ath.c madwifi-0.9.2-2619/ath/if_ath.c
--- madwifi-0.9.2/ath/if_ath.c	2006-07-08 08:47:19.000000000 +0200
+++ madwifi-0.9.2-2619/ath/if_ath.c	2006-11-30 18:11:08.000000000 +0100
@@ -44,8 +44,8 @@
  */
 #include "opt_ah.h"
 
-#include <linux/config.h>
 #include <linux/version.h>
+#include <linux/utsrelease.h>
 #include <linux/module.h>
 #include <linux/init.h>
 #include <linux/skbuff.h>
@@ -1581,7 +1581,11 @@
  * Interrupt handler.  Most of the actual processing is deferred.
  */
 irqreturn_t
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19))
+ath_intr(int irq, void *dev_id)
+#else
 ath_intr(int irq, void *dev_id, struct pt_regs *regs)
+#endif
 {
 	struct net_device *dev = dev_id;
 	struct ath_softc *sc = dev->priv;
diff -Naur madwifi-0.9.2/ath/if_ath_pci.c madwifi-0.9.2-2619/ath/if_ath_pci.c
--- madwifi-0.9.2/ath/if_ath_pci.c	2006-07-21 10:00:32.000000000 +0200
+++ madwifi-0.9.2-2619/ath/if_ath_pci.c	2006-11-30 18:03:31.000000000 +0100
@@ -42,7 +42,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -Naur madwifi-0.9.2/ath/if_athvar.h madwifi-0.9.2-2619/ath/if_athvar.h
--- madwifi-0.9.2/ath/if_athvar.h	2006-06-28 08:45:43.000000000 +0200
+++ madwifi-0.9.2-2619/ath/if_athvar.h	2006-11-30 18:10:39.000000000 +0100
@@ -42,6 +42,8 @@
 #ifndef _DEV_ATH_ATHVAR_H
 #define _DEV_ATH_ATHVAR_H
 
+#include <linux/utsrelease.h>
+
 #include "ah.h"
 #include "if_athioctl.h"
 #include "net80211/ieee80211.h"		/* XXX for WME_NUM_AC */
@@ -686,7 +688,11 @@
 void ath_resume(struct net_device *);
 void ath_suspend(struct net_device *);
 void ath_shutdown(struct net_device *);
-irqreturn_t ath_intr(int, void *, struct pt_regs *);
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19))
+irqreturn_t ath_intr(int, void *);
+#else
+irqreturn_t ath_intr(int, void *, struct pt_regs *regs);
+#endif
 int ath_ioctl_ethtool(struct ath_softc *, int, void __user *);
 void bus_read_cachesize(struct ath_softc *, u_int8_t *);
 #ifdef CONFIG_SYSCTL
diff -Naur madwifi-0.9.2/ath_rate/sample/sample.c madwifi-0.9.2-2619/ath_rate/sample/sample.c
--- madwifi-0.9.2/ath_rate/sample/sample.c	2006-07-04 12:23:35.000000000 +0200
+++ madwifi-0.9.2-2619/ath_rate/sample/sample.c	2006-11-30 18:11:38.000000000 +0100
@@ -41,7 +41,6 @@
  * John Bicket's SampleRate control algorithm.
  */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -Naur madwifi-0.9.2/hal/linux/ah_osdep.c madwifi-0.9.2-2619/hal/linux/ah_osdep.c
--- madwifi-0.9.2/hal/linux/ah_osdep.c	2006-05-19 12:25:45.000000000 +0200
+++ madwifi-0.9.2-2619/hal/linux/ah_osdep.c	2006-11-30 18:02:25.000000000 +0100
@@ -41,7 +41,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211.c madwifi-0.9.2-2619/net80211/ieee80211.c
--- madwifi-0.9.2/net80211/ieee80211.c	2006-07-04 12:22:11.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211.c	2006-11-30 18:12:21.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 generic handler
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_acl.c madwifi-0.9.2-2619/net80211/ieee80211_acl.c
--- madwifi-0.9.2/net80211/ieee80211_acl.c	2006-05-31 23:05:08.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_acl.c	2006-11-30 18:16:37.000000000 +0100
@@ -45,7 +45,6 @@
  * and if found the frame is either accepted (ACL_POLICY_ALLOW)
  * or rejected (ACL_POLICY_DENT).
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_beacon.c madwifi-0.9.2-2619/net80211/ieee80211_beacon.c
--- madwifi-0.9.2/net80211/ieee80211_beacon.c	2006-05-19 22:42:19.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_beacon.c	2006-11-30 18:12:39.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 beacon handling routines
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_crypto.c madwifi-0.9.2-2619/net80211/ieee80211_crypto.c
--- madwifi-0.9.2/net80211/ieee80211_crypto.c	2006-02-01 21:07:11.000000000 +0100
+++ madwifi-0.9.2-2619/net80211/ieee80211_crypto.c	2006-11-30 18:12:56.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 generic crypto support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kmod.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_crypto_ccmp.c madwifi-0.9.2-2619/net80211/ieee80211_crypto_ccmp.c
--- madwifi-0.9.2/net80211/ieee80211_crypto_ccmp.c	2006-06-13 16:09:51.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_crypto_ccmp.c	2006-11-30 18:19:55.000000000 +0100
@@ -38,7 +38,6 @@
  * AP driver. The code is used with the consent of the author and
  * it's license is included below.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
@@ -297,6 +296,9 @@
 static void
 rijndael_encrypt(struct crypto_tfm *tfm, const void *src, void *dst)
 {
+#ifdef CRYPTO_ALG_ASYNC
+	crypto_cipher_encrypt_one(tfm, dst, src);
+#else
 	struct scatterlist sg_src;
 	struct scatterlist sg_dst;
 
@@ -308,6 +310,7 @@
 	sg_dst.offset = offset_in_page(dst);
 	sg_dst.length = AES_BLOCK_LEN;
 	crypto_cipher_encrypt(tfm, &sg_dst, &sg_src, AES_BLOCK_LEN);
+#endif
 }
 
 /*
diff -Naur madwifi-0.9.2/net80211/ieee80211_crypto_none.c madwifi-0.9.2-2619/net80211/ieee80211_crypto_none.c
--- madwifi-0.9.2/net80211/ieee80211_crypto_none.c	2006-02-01 21:07:11.000000000 +0100
+++ madwifi-0.9.2-2619/net80211/ieee80211_crypto_none.c	2006-11-30 18:13:12.000000000 +0100
@@ -34,7 +34,6 @@
 /*
  * IEEE 802.11 NULL crypto support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_crypto_tkip.c madwifi-0.9.2-2619/net80211/ieee80211_crypto_tkip.c
--- madwifi-0.9.2/net80211/ieee80211_crypto_tkip.c	2006-02-01 21:07:11.000000000 +0100
+++ madwifi-0.9.2-2619/net80211/ieee80211_crypto_tkip.c	2006-11-30 18:16:37.000000000 +0100
@@ -38,7 +38,6 @@
  * AP driver. The code is used with the consent of the author and
  * it's license is included below.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_crypto_wep.c madwifi-0.9.2-2619/net80211/ieee80211_crypto_wep.c
--- madwifi-0.9.2/net80211/ieee80211_crypto_wep.c	2006-02-06 21:20:57.000000000 +0100
+++ madwifi-0.9.2-2619/net80211/ieee80211_crypto_wep.c	2006-11-30 18:16:37.000000000 +0100
@@ -34,7 +34,6 @@
 /*
  * IEEE 802.11 WEP crypto support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_input.c madwifi-0.9.2-2619/net80211/ieee80211_input.c
--- madwifi-0.9.2/net80211/ieee80211_input.c	2006-07-06 05:23:08.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_input.c	2006-11-30 18:13:28.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 input handling.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_linux.c madwifi-0.9.2-2619/net80211/ieee80211_linux.c
--- madwifi-0.9.2/net80211/ieee80211_linux.c	2006-07-21 10:59:10.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_linux.c	2006-11-30 18:16:37.000000000 +0100
@@ -33,7 +33,6 @@
 /*
  * IEEE 802.11 support (Linux-specific code)
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kmod.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_monitor.c madwifi-0.9.2-2619/net80211/ieee80211_monitor.c
--- madwifi-0.9.2/net80211/ieee80211_monitor.c	2006-04-21 18:57:59.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_monitor.c	2006-11-30 18:16:37.000000000 +0100
@@ -34,7 +34,6 @@
 /*
  * IEEE 802.11 monitor mode 
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kmod.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_node.c madwifi-0.9.2-2619/net80211/ieee80211_node.c
--- madwifi-0.9.2/net80211/ieee80211_node.c	2006-06-13 10:50:37.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_node.c	2006-11-30 18:13:50.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 node handling support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_output.c madwifi-0.9.2-2619/net80211/ieee80211_output.c
--- madwifi-0.9.2/net80211/ieee80211_output.c	2006-06-10 04:17:05.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_output.c	2006-11-30 18:14:10.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 output handling.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_power.c madwifi-0.9.2-2619/net80211/ieee80211_power.c
--- madwifi-0.9.2/net80211/ieee80211_power.c	2006-06-10 04:17:05.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_power.c	2006-11-30 18:15:27.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 power save support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_proto.c madwifi-0.9.2-2619/net80211/ieee80211_proto.c
--- madwifi-0.9.2/net80211/ieee80211_proto.c	2006-06-09 22:41:19.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_proto.c	2006-11-30 18:15:47.000000000 +0100
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 protocol support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/kmod.h>
 #include <linux/module.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_scan.c madwifi-0.9.2-2619/net80211/ieee80211_scan.c
--- madwifi-0.9.2/net80211/ieee80211_scan.c	2006-04-21 18:57:59.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_scan.c	2006-11-30 18:16:37.000000000 +0100
@@ -37,7 +37,6 @@
 /*
  * IEEE 802.11 scanning support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_scan_ap.c madwifi-0.9.2-2619/net80211/ieee80211_scan_ap.c
--- madwifi-0.9.2/net80211/ieee80211_scan_ap.c	2006-02-01 21:07:11.000000000 +0100
+++ madwifi-0.9.2-2619/net80211/ieee80211_scan_ap.c	2006-11-30 18:16:37.000000000 +0100
@@ -37,7 +37,6 @@
 /*
  * IEEE 802.11 ap scanning support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_scan_sta.c madwifi-0.9.2-2619/net80211/ieee80211_scan_sta.c
--- madwifi-0.9.2/net80211/ieee80211_scan_sta.c	2006-07-02 09:19:37.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_scan_sta.c	2006-11-30 18:16:37.000000000 +0100
@@ -37,7 +37,6 @@
 /*
  * IEEE 802.11 station scanning support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_wireless.c madwifi-0.9.2-2619/net80211/ieee80211_wireless.c
--- madwifi-0.9.2/net80211/ieee80211_wireless.c	2006-07-06 05:23:08.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/ieee80211_wireless.c	2006-11-30 18:16:37.000000000 +0100
@@ -39,7 +39,6 @@
 /*
  * Wireless extensions support for 802.11 common code.
  */
-#include <linux/config.h>
 
 #ifdef CONFIG_NET_WIRELESS
 #include <linux/version.h>
diff -Naur madwifi-0.9.2/net80211/ieee80211_xauth.c madwifi-0.9.2-2619/net80211/ieee80211_xauth.c
--- madwifi-0.9.2/net80211/ieee80211_xauth.c	2006-02-01 21:07:11.000000000 +0100
+++ madwifi-0.9.2-2619/net80211/ieee80211_xauth.c	2006-11-30 18:16:37.000000000 +0100
@@ -46,7 +46,6 @@
  * of the available callbacks--the user mode authenticator process works
  * entirely from messages about stations joining and leaving.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -Naur madwifi-0.9.2/net80211/if_media.c madwifi-0.9.2-2619/net80211/if_media.c
--- madwifi-0.9.2/net80211/if_media.c	2006-05-31 23:05:08.000000000 +0200
+++ madwifi-0.9.2-2619/net80211/if_media.c	2006-11-30 18:12:04.000000000 +0100
@@ -49,7 +49,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
